src/components/Menus/Menu.js

```diff
@@ -0,0 +1,204 @@
+    onBlur: PropTypes.func,
+    /** Role HTML attribute */
+    role: PropTypes.string,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    role: 'menu',
+    maxHeight: 500,
+    ariaHidden: false,
+    style: {},
+  }
+
+  state = {
+    currentTabIndex: null
+  }
+
+  _handleClick = (option, index) => {
+    const { onSelect } = this.props
+
+    onSelect && onSelect(option, index)
+  }
+
+  _handleBlur = (event) => {
+    const { onBlur } = this.props
+    const currentTarget = event.currentTarget
+
+    setTimeout( () => {
+      if (!currentTarget.contains(document.activeElement)) {
+        this.setState({currentTabIndex: null}, () => {
+          onBlur && onBlur(event)
+        })
+      }
+    }, 0)
+  }
+
+  _handleKeyDown = (event) => {
+    const { onKeyDown } = this.props
+
+    switch(event.key) {
+      case 'ArrowDown':
+        event.preventDefault()
+        this.incrementTabIndex()
+        break
+      case 'ArrowUp':
+        event.preventDefault()
+        this.decrementTabIndex()
+        break
+    }
+
+    onKeyDown && onKeyDown(event)
+  }
+
+  _handleMenuItemFocus = (index) => {
+    this.setState({currentTabIndex: index})
+  }
+
+  blur() {
+    this.menu.blur()
+  }
+
+  focus() {
+    this.menu.focus()
+  }
+
+  incrementTabIndex() {
+    this.setState({currentTabIndex: this.nextValidTabIndex()})
+  }
+
+  decrementTabIndex() {
+    this.setState({currentTabIndex: this.prevValidTabIndex()})
+  }
+
+  nextValidTabIndex() {
+    const { currentTabIndex } = this.state
+    const menuItemChildren = this.getMenuItemChildren()
+    const maxIndex = menuItemChildren.length - 1
+    const newIndex = currentTabIndex !== null ? currentTabIndex+1 : 0
+
+    for (let index = newIndex; index <= maxIndex; index++) {
+      const menuItem = menuItemChildren[index]
+
+      if (this.menuItemIsValid(menuItem)) {
+        return index
+      }
+    }
+
+    return currentTabIndex
+  }
+
+  prevValidTabIndex() {
+    const { currentTabIndex } = this.state
+    const menuItemChildren = this.getMenuItemChildren()
+
+    for (let index = (currentTabIndex-1); index >= 0; index--) {
+      const menuItem = menuItemChildren[index]
+
+      if (this.menuItemIsValid(menuItem)) {
+        return index
+      }
+    }
+
+    return currentTabIndex
+  }
+
+  menuItemIsValid(menuItem) {
+    return menuItem && !menuItem.props.disabled
+  }
+
+  getMenuItemChildren() {
+    const { children } = this.props
+    const menuItemChildren = []
+
+    React.Children.map(children, (child) => {
+      if (child.type && child.type.displayName === 'MenuItem') {
+        menuItemChildren.push(child)
+      }
+    })
+
+    return menuItemChildren
+  }
+
+  renderChildren() {
+    const { children } = this.props
+    const { currentTabIndex } = this.state
+    let index = 0
+
+    return React.Children.map(children, (child) => {
+      if (!React.isValidElement(child)) {
```

Comment:
I think we should fail hard here, if it's not a valid element the developer should catch it in development